src/OXRS_LCD.cpp

```diff
@@ -563,6 +590,81 @@ void OXRS_LCD::_update_output_128 (uint8_t type, uint8_t index, int active)
   }     
 }
 
+/**
+  animation of input and outputstate in ports view
```

Comment:
This looks great, but I am worried we are building too much firmware-specific logic into the core LCD library. It feels like this library should be pretty generic and provide tools for the calling firmware to display whatever it needs.

We don't know what the next project will need and we will likely end up having to add new layouts for each one - and this is not a job for the core OXRS team :).

Is there a way we could re-architect the LCD library so you tell it how many ports to display, and then during an event you just pass a port number, a channel number (1-4), a type (input or output), and a state.

The library would become very "dumb" so-to-speak, and the firmware would decide what and when to show something. Then it can remain simple and won't need to be touched as new projects come on board.

Thoughts?

Reply:
I agree that we need kind of a standard for the port animation.

As of now we have 3 different basic layouts:
- inputs
- outputs
- I/O combined
This should be defined by the user (FW-SW) during `draw_ports(..) `so that the port layout is defined. An additional parameter could be used to define the number of ports/mcps that should be shown (defaults to max). With these 2 parameter the port layout will be rendered.

The animation is based on what is read from the MCPs, independet from any event/cmnd via MQTT. The animation is meant to be seen as if there would be and LED connected to the MCP I/O pin. The animation then is triggered by `process (..)` which inturn evaluates the io_value read.

I think this is pretty much what we can do. The basic layout needs to be served by the lib.

I hope I got your comment right?

For now I'll merge this PR .